readers/api/transport.go

```diff
@@ -229,23 +229,23 @@ func authorize(ctx context.Context, r *http.Request, chanID string) (err error)
 		if err != nil {
 			e, ok := status.FromError(err)
 			if ok && e.Code() == codes.PermissionDenied {
-				return errUnauthorizedAccess
+				return errCannotAuthorizeUser
 			}
-			return errors.Wrap(errUnauthorizedAccess, errCannotAuthorizeUser)
+			return errCannotAuthorizeUser
```

Comment:
You are always returning `errCannotAuthorizeUser`. Do we need to check if `e.Code() == codes.PermissionDenied`? Also, can we use `errors.ErrAuthorization` instead of the custom one?